From 1ac45d880204e0a6fc59900324bc12cc37d09de9 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 07 Jun 2022 07:02:15 +0300
Subject: [PATCH] [SW-0] pcie: set device id after prstn is deasserted

As per arch team suggestion, setting device id for slave die is moved
after prstn is deasserted.

Change-Id: I4051b75abc264db51a9e8a134e85ca431d8bf53f
---

diff --git a/common/pcie.c b/common/pcie.c
index 92c55a4..83b0fc7 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -80,9 +80,10 @@
 	uint32_t link_ctrl;
 	uint8_t ltssm_en;
 
+	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
+
 	pcie_set_pcie_device_vendor_id();
 
-	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
 	ltssm_en = regmap_readl(mmLTSSM_EN) &
 				LTSSM_EN_IND_MASK;
 
